### PR TITLE
Fix linting and formatting issues

### PR DESCRIPTION
**Pull request by adrianosela**
Trying to get featured in https://github.com/avelino/awesome-go

The repo's maintainer has asked we take a look at our [go report card topics](https://goreportcard.com/report/github.com/mozilla-services/sops)

I've remediated all entries which do not require a significant change of code, it is mostly just the addition of comments to exported functions. 

### Note3: A report card topic which I have not addressed are cyclomatic complexities, which I too have left out on purpose for a subsequent PR.

### CODE REVIEW

File: cmd/sops/common/common.go [hunk 1]
```diff
@@ -143,10 +154,14 @@ func DefaultStoreForPath(path string) Store {
 	return &json.BinaryStore{}
 }
 
-const KMS_ENC_CTX_BUG_FIXED_VERSION = "3.3.0"
+// KmsEncCtxBugFixedVersion represents the SOPS version in which the
+// encryption context bug was fixed
+const KmsEncCtxBugFixedVersion = "3.3.0"
```
Comment (ajvb) [inline cmd/sops/common/common.go]: This should stay as `KMS_ENC_CTX_BUG_FIXED_VERSION`

File: shamir/shamir.go [hunk 2]
```diff
@@ -127,16 +127,16 @@ func div(a, b uint8) uint8 {
 // mult multiplies two numbers in GF(2^8)
 // GF(2^8) multiplication using log/exp tables
 func mult(a, b uint8) (out uint8) {
-	var goodVal, zero uint8
-	log_a := logTable[a]
-	log_b := logTable[b]
-	sum := (int(log_a) + int(log_b)) % 255
+	var zero uint8
```
Comment (ajvb) [inline shamir/shamir.go]: Let's back out all changes in this file except those relating to renaming variables.

Comment (adrianosela) [reply shamir/shamir.go]: ✅

File: cmd/sops/common/common.go (hunk 1 above)
Comment (adrianosela) [reply cmd/sops/common/common.go]: ✅

**Pull request merged**